vllm/worker/hpu_model_runner.py

```diff
@@ -1976,6 +2211,13 @@ def prepare_model_input(
                     seq_group_metadata_list=seq_group_metadata_list)
             model_input, sampling_metadata = self.prepare_input_tensors(
                 seq_group_metadata_list)
+            if self.is_encoder_decoder_model:
```

Comment:
`prepare_input_tensors` is still necessary for encoder_decoder model?

Reply:
Reference to other HW, they added a new model_runner, ex. cpu_encoder_decoder_model_runner, then we can put
`prepare_input_tensors` and `HPUAdapter` in that file to reduce changes to "hpu_model_runner.py"
Should we do same thing here?
```
class CPUEncoderDecoderModelRunner(
        CPUModelRunnerBase[EncoderDecoderModelInputForCPU]):
```

Reply:
@xuechendi  Yes, I plan to create `encoder_decoder_model_runner` in the next PR that support var-len image inputs.

Reply:
@jikunshang Yes, `prepare_input_tensors` is still necessary for the encoder_decoder model, as it adds the encoder-cross layer while retaining the decoder layers. The `prepare_input_tensors` is dedicated to build decoder layer inputs.

Reply:
@yisonzhu , I am thinking can we create a HPU_encoder_decoder_model_runner in this PR? Since this is in your plan, we should keep minimum change to hpu_model_runner since it is on critical path.

Reply:
OK, make sense. I will create HPU_encoder_decoder_model_runner in this PR. Meanwhile, profile_run can be also supported using this design.

Reply:
@xuechendi I have done the refactor and support warm-up buckets for encoder inputs. I also collect some performance data in the description above. Could you please make a review again?